code/modules/mob/living/silicon/robot/robot_model.dm

```diff
@@ -129,7 +132,7 @@
 	for(var/obj/item/module in added_modules)
 		add_module(module, FALSE, FALSE)
 	for(var/module in held_modules)
```

Comment:
also, consider
```suggestion
	for(var/module in held_modules & modules)
```
also what is module? could you please type it correctly

Reply:
'held_modules' is a number indexed array of modules and 'modules' is a set of modules. would the '&' operator work for that combination?

Reply:
I tried it out and it looks to have worked

Reply:
ok so just to be clear there's only one list type in byond.
lists can hold values, += and all.
lists can also map values to other values, typically these are done separately. 

maps (assoc lists) are still just lists. The keys are just values stored sequentially in the list, we just also use them to look up in a tree stored internally.

Reply:
> 'held_modules' is a number indexed array of modules and 'modules' is a set of modules. would the '&' operator work for that combination?

In context of the above, yes. Assuming held_modules is a list of held modules and modules is a list of all possible modules, it will filter for only entries in both lists.

Reply:
gotcha